src/parser.ts

```diff
@@ -355,10 +355,20 @@ const EXPR_BUILDER: MarkVisitor<NodeTypes.ExprNode> = {
       return result
     }
 
-    const args: NodeTypes.ExprNode[] = []
+    let args: NodeTypes.ExprNode[] = []
+
     while (p.getMark().name !== 'func_args_end') {
-      args.push(p.process(EXPR_BUILDER))
+      if (argumentShouldBeSelector(namespace, name, args)) {
```

Comment:
Ah, this code confused me first. The reason this works is because we populate the `args` on the fly and thus it contains all the previous arguments. It's rather strange for the `argumentShouldBeSelector`-function to get access to all of the _previous_ arguments though. What about changing this to:
```suggestion
      if (argumentShouldBeSelector(namespace, name, args.length)) {
```

And then the `argumentShouldBeSelector` actually accepts an `argumentIndex`?